### PR TITLE
Attachment parameters and properties MUST match!

### LINKED ISSUE #513: Attachments: "contentType" must match part's "Content-Type"
The spec in 4.1.11 says:

> SHOULD include a Content-type field in each part's header, for the first part this MUST be "application/json".

And we were discussing whether there should be a MUST added such that it states that the content type in the headers of the part *MUST* match the content type as is provided in the statement metadata. There isn't really any other rational choice as far as I know, but being explicit about it, well, makes it explicit.

### PR DESCRIPTION
**Pull request by garemoko**
Fixes #513 

The issue also applied to content-length so I fixed this in a more generic way, by adding an extra column to the table to match parameters to properties and saying that they always have to match.

### CODE REVIEW

**Comment by garemoko** [comment]
It's probably worthwhile looking at the rich diff of this one for clarity.

File: xAPI.md [hunk 1]
```diff
@@ -1970,13 +1977,13 @@ results when the attachments filter is false.
...
-	* MUST include an X-Experience-API-Hash parameter in each part's header after the first (Statements) part.
-	* This parameter MUST be set to match the "sha2" property of the attachment declaration corresponding to the 
-	attachment included in this part.
-	* MUST include a Content-Transfer-Encoding parameter with a value of "binary" in each part's header after the first (statements) part.
+    capability will be available when issuing PUT or POST against the Statement resource.
+    * MUST include an X-Experience-API-Hash parameter in each part's header after the first (Statements) part.
+    * MUST include a Content-Transfer-Encoding parameter with a value of "binary" in each part's header after the first (statements) part.
     * SHOULD only include one copy of an attachment's data when the same attachment is used in multiple Statements that are sent together.
-    * SHOULD include a Content-Type parameter in each part's header, for the first part this MUST be "application/json".
+    * SHOULD include a Content-Type parameter in each part's header, for the first part (containing the statement) this MUST be "application/json".
```
Comment (fugu13) [inline xAPI.md]: How about period instead of comma, to prevent accidental misreadings

File: xAPI.md [hunk 2]
```diff
@@ -1970,13 +1977,13 @@ results when the attachments filter is false.
...
+    * Where parameters have a corresponding property within the Attachment Object (outlined in the table above), the value of these parameters and properties
+    for each Attachment MUST match.
```
Comment (fugu13) [inline xAPI.md]: Lets finish with "if specified" so people are sure they can leave content-type headers out.

File: xAPI.md (hunk 1 above)
Comment (garemoko) [inline xAPI.md]: +1

**Comment by andyjohnson** [comment]
+1

File: xAPI.md [hunk 3]
```diff
@@ -1970,13 +1977,13 @@ results when the attachments filter is false.
...
-	attachment included in this part.
-	* MUST include a Content-Transfer-Encoding parameter with a value of "binary" in each part's header after the first (statements) part.
+    capability will be available when issuing PUT or POST against the Statement resource.
+    * MUST include an X-Experience-API-Hash parameter in each part's header after the first (Statements) part.
+    * MUST include a Content-Transfer-Encoding parameter with a value of "binary" in each part's header after the first (statements) part.
     * SHOULD only include one copy of an attachment's data when the same attachment is used in multiple Statements that are sent together.
-    * SHOULD include a Content-Type parameter in each part's header, for the first part this MUST be "application/json".
+    * SHOULD include a Content-Type parameter in each part's header. For the first part (containing the statement) this MUST be "application/json".
+    * Where parameters have a corresponding property within the Attachment Object (outlined in the table above), the value of these parameters and properties
+    for each Attachment MUST match.
```
Comment (fugu13) [inline xAPI.md]: Looks like an edit hid an earlier comment of mine:

Lets finish with "if specified" so people are sure they can leave content-type headers out.

Comment (garemoko) [inline xAPI.md]: How about: ```* Where parameters have a corresponding property within the Attachment Object (outlined in the table above), and both the parameter and property are specified for a given Attachment, the value of these parameters and properties MUST match.```

Comment (fugu13) [inline xAPI.md]: Yeah, that's better

**Comment by garemoko** [comment]
I believe this addresses all of @fugu13's concerns.

**Comment by ljwolford** [comment]
+1

**Pull request merged**